azurerm/data_source_compute_resource_sku.go

```diff
@@ -0,0 +1,442 @@
+						},
+
+						"zones": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+
+			"api_versions": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"costs": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"meter_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"quantity": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+
+						"extended_unit": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"capabilities": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"value": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"restrictions": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"restriction_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"values": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+
+						// TODO
+						"restriction_info": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"locations": {
+										Type:     schema.TypeList,
+										Computed: true,
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+										},
+									},
+
+									"zones": {
+										Type:     schema.TypeList,
+										Computed: true,
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+										},
+									},
+								},
+							},
+						},
+
+						"reason_code": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceArmComputeResourceSkuRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).compute.ResourceSkusClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+
+	resourceSkus, err := client.ListComplete(ctx)
+
+	if err != nil {
+		return fmt.Errorf("Error loading Compute Resource SKUs: %+v", err)
+	}
+
+	filteredResourceSkus := make([]compute.ResourceSku, 0)
+	for resourceSkus.NotDone() {
+		shouldInclude := false
+		sku := resourceSkus.Value()
+
+		if v, ok := d.GetOkExists("location"); ok {
+			if location := v.(string); location != "" {
+				if Contains(*sku.Locations, location) && *sku.Name == name {
+					shouldInclude = true
```

Comment:
Just clarifying because I'm not 100% positive this is what you meant... but like this?

``` golang
	name := d.Get("name").(string)
	location := d.Get("location").(string)

	resourceSkus, err := client.ListComplete(ctx)

	if err != nil {
		return fmt.Errorf("Error loading Compute Resource SKUs: %+v", err)
	}
        for resourceSkus.NotDone() {
		// shouldInclude := false
		// sku := resourceSkus.Value()

		// if v, ok := d.GetOkExists("location"); ok {
		// 	if location := v.(string); location != "" {
		// 		if Contains(*sku.Locations, location) && *sku.Name == name {
		// 			shouldInclude = true
		// 		}
		// 	}
		// } else {
		// 	if *sku.Name == name {
		// 		shouldInclude = true
		// 	}
		// }

		// if shouldInclude {
		// 	filteredResourceSkus = append(filteredResourceSkus, sku)
		// }

		sku := resourceSkus.Value()
		filteredResourceSkus = append(filteredResourceSkus, sku)
		err = resourceSkus.NextWithContext(ctx)
		if err != nil {
			return fmt.Errorf("Error loading Compute Resource SKUs: %+v", err)
		}
	}
```

Assuming this is correct, I'll remove the commented lines.